Snippets/Snippets_5/Features/CustomFaultManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace Snippets5.Features
+{
+    using System;
+    using NServiceBus;
+    using NServiceBus.Faults;
+
+    #region CustomFaultManager
+    public class CustomFaultManager : IManageMessageFailures
+    {
+        public void SerializationFailedForMessage(TransportMessage message, Exception e)
+        {
+            // implement any custom steps for this message when the failure was due to deserialization
+        }
+
+        public void ProcessingAlwaysFailsForMessage(TransportMessage message, Exception e)
+        {
```

Comment:
@weralabaj - For V5, SLR won't be invoked. When you implement this extension, you're taking over the fault managing process. I've clarified this better now.